blackbird_python/blackbird/program.py

```diff
@@ -401,7 +425,12 @@ def serialize(self):
                         array_insert += len(bb_array)
 
                     elif isinstance(v, str):
-                        kwargs.append('{}="{}"'.format(k, v))
+                        # kwarg is a string type; if a p-type parameter (e.g. p0),
+                        # then simply add it as is
+                        if self.programtype["name"] == "tdm" and v[0] == "p" and v[1:].isdigit():
+                            kwargs.append("{}={}".format(k, v))
+                        else:
+                            kwargs.append('{}="{}"'.format(k, v))
```

Comment:
(Minor suggestion): the difference between the two expressions is minor. It could help the reader if the value was extracted somehow to show the difference (not sure if this the suggestion is much better though :sweat_smile:)
```python
                        if self.programtype["name"] == "tdm" and v[0] == "p" and v[1:].isdigit():
                            value = "{}".format(v)
                        else:
                            value = '"{}"'.format(v)
                        kwargs.append("{}={}".format(k, value))
```

Reply:
Thanks @antalszava! I might just keep this as it is now. I think it's clear enough, although I see your point. :slightly_smiling_face: